addons/spreadsheet_account/models/account.py

```diff
@@ -79,50 +71,282 @@ def _build_spreadsheet_formula_domain(self, formula_params):
             domain = expression.AND(
                 [domain, [("move_id.state", "=", "posted")]]
             )
+        partner_ids = [int(partner_id) for partner_id in formula_params.get("partner_id", []) if partner_id]
+        if partner_ids:
+            domain = expression.AND(
+                [domain, [("partner_id", "in", partner_ids)]]
+            )
         return domain
 
+    def _pre_process_date_period_boundaries(self, args_list):
+        for args in args_list:
+            company_id = args['company_id'] or self.env.company.id
+            company = self.env['res.company'].browse(company_id)
+
+            start, _end = self._get_date_period_boundaries(
+                args['date_from'], company
+            )
+            _start, end = self._get_date_period_boundaries(
+                args['date_to'], company
+            )
+
+            args['date_from_boundary'] = start
+            args['date_to_boundary'] = end
+
+    def _pre_process_timeline(self, args_list):
```

Comment:
You have another `_pre_process` that doesn't behave the same way, the nomenclature doesn't make sense.
```suggestion
    def _get_timelines(self, args_list):
```